### PR TITLE
0.13 fix: use default target in sync overrides

### PR DESCRIPTION
**Pull request by Orzelius**
If a target is not specified for a sync override the default target is used.

Also unified the schema and types between targets for sync and sync overrides.

Changed the behavior to throw if target is not set instead of a warning. I don't think it makes sense to just warn here.

### CODE REVIEW

File: core/src/plugins/kubernetes/sync.ts [hunk 1]
```diff
@@ -204,7 +193,10 @@ export const kubernetesDeploySyncSchema = () =>
         .array()
         .items(kubernetesDeploySyncPathSchema())
         .description("A list of syncs to start once the Deploy is successfully started."),
-      overrides: joi.array().items(syncModeOverrideSpec()),
+      overrides: joi
+        .array()
+        .items(syncModeOverrideSpec())
+        .description("Overrides for the container command and/or arguments for when in sync mode"),
```
Comment (vvagaytsev) [inline core/src/plugins/kubernetes/sync.ts:199]:
```suggestion
        .description("Overrides for the container command and/or arguments for when in sync mode"),
        .description("Overrides for the container command and/or arguments when in sync mode."),
```

Comment (vvagaytsev) [reply core/src/plugins/kubernetes/sync.ts:199]: Nit: the period is missing at the end of the sentence :) To sure about "for", maybe it's fine to have it there.

File: core/src/plugins/kubernetes/commands/sync.ts [hunk 2]
```diff
@@ -30,9 +30,6 @@ export const syncStatus: PluginCommand = {
     if (!(await pathExists(dataDir))) {
       log.info(dedent`
         No active sync session found.
-
-        Garden needs to be running in dev mode in this project for sync statuses to
```
Comment (vvagaytsev) [inline core/src/plugins/kubernetes/commands/sync.ts:34]: I'm not sure if we should remove this completely. Maybe, we should just replace "dev mode" with "sync mode"

Comment (edvald) [reply core/src/plugins/kubernetes/commands/sync.ts:34]: I think this is fine actually.

File: core/src/plugins/kubernetes/sync.ts (hunk 1 above)
Comment (edvald) [reply core/src/plugins/kubernetes/sync.ts:199]: 👍

File: core/src/plugins/kubernetes/sync.ts [hunk 3]
```diff
@@ -332,7 +325,15 @@ export async function configureSyncMode({
   }
 
   for (const override of spec.overrides || []) {
-    const { target } = override
+    const target = override.target || defaultTarget
+    if (!target) {
+      throw new ConfigurationError(
+        `Sync override configuration on ${action.longDescription()} doesn't specify a target, and none is set as a default.`,
```
Comment (edvald) [inline core/src/plugins/kubernetes/sync.ts:331]: Maybe we can provide more clear instructions here, i.e. which field needs to be filled in.

**Review by edvald — CHANGES_REQUESTED** [review]
Suggested improving the error message, otherwise looks good!

**Comment by vvagaytsev** [comment]
The [check-docs](https://app.circleci.com/pipelines/github/garden-io/garden/16517/workflows/9e77eead-2805-4403-98b5-3bae6d39fd2c/jobs/279717) step failed. Please, re-generate the docs.

**Review by vvagaytsev — APPROVED** [review]

**Pull request merged**